service/service.go

```diff
@@ -146,3 +146,19 @@ func New(options ...Option) (Owner, error) {
 
 	return svc, nil
 }
+
+type niladicStart func()
+
+func (n niladicStart) OnInit(service Host) error      { return nil }
+func (n niladicStart) OnShutdown(reason Exit)         {}
+func (n niladicStart) OnCriticalError(err error) bool { return true }
+func (n niladicStart) OnStateChange(old State, new State) {
+	if old == Starting && new == Running {
```

Comment:
We do transition from [Starting](https://github.com/uber-go/fx/blob/master/service/host.go#L235) to [Running](https://github.com/uber-go/fx/blob/master/service/host.go#L297) in start